lib/webdriveragent.js

```diff
@@ -80,12 +110,15 @@ class WebDriverAgent {
     if (this.webDriverAgentUrl) {
       log.info(`Using provided WebdriverAgent at '${this.webDriverAgentUrl}'`);
       this.url = this.webDriverAgentUrl;
+      this.setupNoSessionProxy();
       this.setupProxy(sessionId);
       return this.webDriverAgentUrl;
     }
 
     log.info('Launching WebDriverAgent on the device');
 
+    this.setupNoSessionProxy();
```

Comment:
it's not going to be called twice, since we have return in the condition above. But yes, nothing prevents us to move this method before if condition.